burger-app/src/contents/Main/mainContent.js

```diff
@@ -0,0 +1,142 @@
+import styled from "styled-components";
+import React from "react";
+import { Price, Burger, Controls } from "../../components/BodyMain";
+
+class Main extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      prices: [],
+      ingredients: [],
+      ingredientAddingOrder: [],
+      burgerCreator: {
+        meat: 0,
+        pickle: 0,
+        salad: 0,
+        chicken: 0,
+        cheese: 0,
+        tomato: 0,
+        onion: 0,
+      },
+      orderPrice: "1.00",
+    };
+  }
+
+  componentDidMount = async () => {
+    const data = await await fetch(
+      "https://burger-api-xcwp.onrender.com/ingredients"
+    ).then((res) => res.json());
+
+    const ingredients = data.map((ingredient) => {
+      return ingredient.name;
+    });
+
+    const quantities = data.reduce(
+      (acc, curr) => ({ [curr.name]: 0, ...acc }),
+      {}
+    );
+
+    this.setState({
+      prices: data,
+      ingredients: ingredients,
+      burgerCreator: quantities,
+    });
+  };
+
+  findIngredientPrice = (ingredient) => {
+    return this.state.prices.find((price) => price.name === ingredient).price;
+  };
+
+  handleChangeBurgerIngredientQuantity = (event) => {
```

Comment:
The function is almost identical to our burger app. Would be better to finalize it and create own logic 😢